shadow.c

```diff
@@ -549,15 +559,29 @@ static char *get_full_path(const char *filename TSRMLS_DC)
 		SHADOW_G(curdir) = getcwd(NULL, 0);
 	}
 
+#if PHP_VERSION_ID < 50600
 	new_state.cwd = strdup(SHADOW_G(curdir));
+#else
+	new_state.cwd = estrdup(SHADOW_G(curdir));
```

Comment:
@morozov , here's the commit in 5.6 that fixes it - https://github.com/php/php-src/commit/cf6ab0e915f95f01a8db1ce7daf70ed36cf49c66
I specified this link in my commit message for reference.

@yellowandy , I've added macro functions to make it look cleaner